sql/table.go

```diff
@@ -164,6 +164,8 @@ func encodeTableKey(b []byte, v parser.Datum) ([]byte, error) {
 		return encoding.EncodeNumericFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeBytes(b, []byte(t)), nil
+	case parser.DNull:
+		return b, nil
```

Comment:
We need access to the descriptor anyways as we need to convert the passed in value to the column type. Note https://sqlite.org/src4/doc/trunk/www/key_encoding.wiki. `NULL` gets encoded as a special value. I think we can accomplish something similar with our encoding routines such that `NULL` sorts less than any other encoded value. I'm not sure if all our encoding routines support this yet, but it shouldn't be difficult to modify them to do so.